api/lang/document.go

```diff
@@ -83,13 +94,57 @@ func documentClosed(req *jsonrpc2.Request, ctx context.Context, conn jsonrpc2.JS
 	}
 	if util.IsGaugeFile(string(params.TextDocument.URI)) {
 		closeFile(params)
-		if !common.FileExists(string(util.ConvertPathToURI(params.TextDocument.URI))) {
-			publishDiagnostic(params.TextDocument.URI, []lsp.Diagnostic{}, conn, ctx)
-		}
 	} else if lRunner.runner != nil {
-		cacheFileRequest := &gm.Message{MessageType: gm.Message_CacheFileRequest, CacheFileRequest: &gm.CacheFileRequest{FilePath: string(util.ConvertURItoFilePath(params.TextDocument.URI)), IsClosed: true}}
+		cacheFileRequest := &gm.Message{MessageType: gm.Message_CacheFileRequest, CacheFileRequest: &gm.CacheFileRequest{
+			FilePath: string(util.ConvertURItoFilePath(params.TextDocument.URI)),
+			IsClosed: true,
+			Status:   gm.CacheFileRequest_CLOSED,
+		}}
 		err = sendMessageToRunner(cacheFileRequest)
 	}
 	go publishDiagnostics(ctx, conn)
 	return err
 }
+
+func documentCreate(req *jsonrpc2.Request, ctx context.Context, conn jsonrpc2.JSONRPC2) error {
+	var params lsp.TextDocumentIdentifier
+	var err error
+	if err = json.Unmarshal(*req.Params, &params); err != nil {
+		return fmt.Errorf("failed to parse request %v", err)
```

Comment:
let's not do `%v`, the error messages are unreadable since it brings field names etc. `%s` should be good enough.